src/main/java/ch/cern/eam/wshub/core/services/contractmanagement/impl/EquipmentReservationAdjustmentServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+        return updateOperationResult.getResultData().getCUSTOMERRENTALADJUSTMENTID().getCUSTOMERRENTALADJUSTMENTPK();
+    }
+
+    @Override
+    public String deleteEquipmentReservationAdjustment(InforContext context, String number) throws InforException {
+        MP7866_DeleteCustomerRentalAdjustment_001 deleteOperation = createDeleteOperation(number);
+        MP7866_DeleteCustomerRentalAdjustment_001_Result deleteOperationResult = tools.performInforOperation(context, inforws::deleteCustomerRentalAdjustmentOp, deleteOperation);
+        return deleteOperationResult.getResultData().getCUSTOMERRENTALADJUSTMENTID().getCUSTOMERRENTALADJUSTMENTPK();
+    }
+
+    private MP7864_AddCustomerRentalAdjustment_001 createAddOperation(InforContext context, EquipmentReservationAdjustment equipmentReservationAdjustment) throws InforException {
+        MP7864_AddCustomerRentalAdjustment_001 addCustomerRentalAdjustment = new MP7864_AddCustomerRentalAdjustment_001();
+        CustomerRentalAdjustment customerRentalAdjustment = tools.getInforFieldTools().transformWSHubObject(createDefaultEquipmentReservationAdjustment(), equipmentReservationAdjustment, context);
+        addCustomerRentalAdjustment.setCustomerRentalAdjustment(customerRentalAdjustment);
+        return addCustomerRentalAdjustment;
+    }
+
+    private MP7863_GetCustomerRentalAdjustment_001 createGetOperation(String number) {
+        MP7863_GetCustomerRentalAdjustment_001 getCustomerRentalAdjustment = new MP7863_GetCustomerRentalAdjustment_001();
+        CUSTOMERRENTALADJUSTMENTID_Type idType = new CUSTOMERRENTALADJUSTMENTID_Type();
+        idType.setCUSTOMERRENTALADJUSTMENTPK(number);
+        getCustomerRentalAdjustment.setCUSTOMERRENTALADJUSTMENTID(idType);
+        return getCustomerRentalAdjustment;
+    }
+
+    private MP7865_SyncCustomerRentalAdjustment_001 createUpdateOperation(InforContext context, EquipmentReservationAdjustment equipmentReservationAdjustment) throws InforException {
+        MP7865_SyncCustomerRentalAdjustment_001 syncCustomerRentalAdjustment = new MP7865_SyncCustomerRentalAdjustment_001();
+        EquipmentReservationAdjustment prevReservation = readEquipmentReservationAdjustment(context, equipmentReservationAdjustment.getCode());
+        CustomerRentalAdjustment prevCustomerRentalAdjustment = tools.getInforFieldTools().transformWSHubObject(new CustomerRentalAdjustment(), prevReservation, context);
+        CustomerRentalAdjustment newCustomerRentalAdjustment = tools.getInforFieldTools().transformWSHubObject(prevCustomerRentalAdjustment, equipmentReservationAdjustment, context);
+        syncCustomerRentalAdjustment.setCustomerRentalAdjustment(newCustomerRentalAdjustment);
+        return syncCustomerRentalAdjustment;
+    }
+
+    private MP7866_DeleteCustomerRentalAdjustment_001 createDeleteOperation(String number) {
```

Comment:
This method is not creating the Operation, just the adapting the object that needs to be used. We shall either change the name or include it in the method calling it